sys/include/net/gnrc/pkt.h

```diff
@@ -200,8 +210,10 @@ static inline gnrc_pktsnip_t *gnrc_pkt_prepend(gnrc_pktsnip_t *pkt,
 static inline gnrc_pktsnip_t *gnrc_pkt_delete(gnrc_pktsnip_t *pkt,
                                               gnrc_pktsnip_t *snip)
 {
-    LL_DELETE(pkt, snip);
-    return pkt;
+    list_node_t list = { .next = (list_node_t *)pkt };
+
+    list_remove(&list, (list_node_t *)snip);
+    return (gnrc_pktsnip_t *)list.next;
```

Comment:
If `list` is so confusing, should I maybe replace this with

```suggestion
    if (snip == pkt) {
        return pkt->next;
    }
    else {
        gnrc_pktsnip_t *prev = gnrc_pkt_prev_snip(pkt, snip);
        
        if (prev) {
            prev->next = snip->next;
        }
        return pkt;
    }
```
?

Reply:
nah this is fine - this doesn't *add* complexity where there was none before